content/gsoc/2024/_index.md

```diff
@@ -32,30 +37,66 @@ We provide a [template for proposals](https://docs.google.com/document/d/1gER-yF
+
+PyBaMM already uses `pytest` with the `nbmake` package to run our example notebooks as tests, but still uses `unittest` for the rest of the tests. `pytest-xdist` is also used to run the tests in parallel, which should end up being faster than the current serial implementation and should be adapted to be used with the new test framework.
 
-Project description
+The student will be expected to look at other popular Python packages to see how they use `pytest` and `hypothesis` to write tests, and then migrate our tests over to use these frameworks. This will involve writing new tests, and also migrating existing tests over to the new framework, making use of and exploring the `pytest` documentation for scouting potential features to be incorporated. The student will also be expected to improve the test coverage of the codebase, and to write tests for any new code that they end up writing.
+
+As a stretch goal, the packaging infrastructure for PyBaMM should be updated to include a `pytest` test suite that can be run by users to check that their installation is working correctly for users who do not have access to a source installation of PyBaMM, i.e., through `pybamm.test()` or similar.
 
 #### Expected outcomes
 
-* Expected outcome 1
+* All test files and modules migrated to use `pytest` in the code base, in a way that is consistent with modern Python software engineering and testing practices – either manually or via automated migration tools
+* Improvisation of test coverage
+* Utilisation of advanced testing techniques based on advanced `pytest` features and `hypothesis` as a framework
+* Deprecate some legacy test invocation methods such as `run-tests.py` in favour of the `pytest` path finding and test case discovery mechanisms
 
 #### Desired skills
 
-* Desired skill 1
-* Desired skill 2
-* Desired skill 3
+* Experience with Python and unit testing (`unittest`, `pytest`, and `hypothesis`) in Python.
+* Some experience with test-driven development, continuous integration, and coverage tools is desirable, but not required.
+* Git version control and GitHub workflow for open-source projects.
 
 ### Difficulty and suitable project length
 
-* Easy/medium/hard. This project is suitable for a 175h/350h project.
+* Medium. This project is suitable for a 175-hour duration.
 
 ### Potential mentors
 
-* [Mentor 1](link to website, GitHub profile, or LinkedIn profile)
+* [Agriya Khetarpal](https://agriyakhetarpal.github.io/)
+* [Saransh Chopra](https://Saransh-cpp.github.io/)
+* [Valentin Sulzer](https://sites.google.com/view/valentinsulzer)
```

Comment:
```suggestion
* [Valentin Sulzer](https://sites.google.com/view/valentinsulzer)
* [Arjun Verma](https://github.com/arjxn-py)
```

Reply:
Feel free to let me know in case it only feels more valid to list just adequate number of people.

Reply:
We can add more people to the list for now, and then remove later based on the addition of more projects